File: Chapter 02/Python/2.7.1 Weighted Decision Model/option.py
from dataclasses import dataclass
from typing import Dict

# A simple structure to hold an option and its scores.
# Scores are from 1 (bad) to 5 (good) for each criterion.
@dataclass
class Option:
    name: str
    scores: Dict[str, int]  # e.g., {"speed": 5, "cost": 1, "simplicity": 4}

File: Chapter 02/JS/section_2_7_1_weighted_decision_model/option.js
/**
 * A simple class to hold an option and its scores.
 * Scores are from 1 (bad) to 5 (good) for each criterion.
 */
class Option {
    constructor(name, scores) {
        this.name = name;
        this.scores = scores; // e.g., { availability: 1, performance: 5, simplicity: 5 }
    }
}

module.exports = Option;